server/src/main/java/org/apache/druid/server/coordinator/LoadQueueTaskMaster.java

```diff
@@ -67,7 +68,7 @@ public LoadQueuePeon giveMePeon(ImmutableDruidServer server)
       return new HttpLoadQueuePeon(server.getURL(), jsonMapper, httpClient, config, peonExec, callbackExec);
     } else {
       return new CuratorLoadQueuePeon(
-          curator,
+          curatorFrameworkProvider.get(),
```

Comment:
Just thinking out loud, doesn't need to be addressed in this PR, it seems like maybe `LoadQueueTaskMaster` maybe needs some sort of peon factory that is set by config so that it doesn't have to care about individual implementations or curators and the like

Reply:
could be, but you know my preference by now :)
I wish people would migrate  to using `HTTP` segment management and that would remain the only way and this would be deleted.